### ISSUE TITLE
Windows Support

### ISSUE THREAD

**Issue by friebetill** [amplitude/Amplitude-Flutter#90]
I develop my app for Android, iOS, Windows, macOS and Linux. 
It would be great if you provide support for Windows.

**Comment by jimpanse1337** [comment]
+1 for us at [unping.com](url) , we're working on a multi-platform app, so it would be super helpful

**Comment by bambubuilder** [comment]
+1

**Comment by AmirhosseinShk** [comment]
+1

**Comment by j-junckes** [comment]
I agree, desktop support would be a great feature for it, I was going to go with Mixpanel first, but then found out their Flutter plugin did not support desktop, so I decided to look at Amplitude, disappointed as well...

**Comment by kmalyshev** [comment]
Flutter ios SDK works pretty well with macos (i use it in my swift app), adding flutter-macos should be easy. However windows is a different story.

Where can we vote for this feature request? It would fully cover flutter apps and native windows too

**Comment by past77** [comment]
+1

**Comment by isinghmitesh** [comment]
+1

**Comment by kmalyshev** [comment]
5 years later the support is still not there. Could someone from the dev team comment if there are any technical issues to implement it? Or is there a lack of enough interest from the community that stops implementing it?

**Comment by chungdaniel** [comment]
Hey @kmalyshev, first of all thank you for using Amplitude! Sorry for the lack of updates on this topic. For macOS, since you mentioned that, you were right - it was pretty straight forward to implement and we have support for macOS  apps as of our v4 SDK.

For windows, there are unfortunately some technical issues for implementation. Under the hood, we ultimately have different platforms pointing to our native SDKs (e.g. for flutter-android, we use our [Amplitude-Kotlin SDK](https://github.com/amplitude/Amplitude-Kotlin). Since we don't have an SDK in C/C++, it is a bit more work for us to try and implement this.

**Comment by venkatareddyboya-highspot** [comment]
+1.

**Comment by synkc** [comment]
> Since we don't have an SDK in C/C++, it is a bit more work for us to try and implement this.

@chungdaniel 
Would it be possible to get a rough timeline for the additional work involved?

**Comment by chungdaniel** [comment]
Hey @synkc, I'm seeing that there's a decent chunk of people interested in Windows on this thread, but unfortunately it isn’t something on our roadmap for the foreseeable future, so I don’t want to give a misleading ETA.

Since this SDK is open source, you’re absolutely welcome to fork it and add Windows support in whatever way best fits your app. If you go that route, we’re happy to share pointers on where to plug things in, how the Flutter layer maps to the native SDKs today, and any gotchas we’ve seen. Even if it stays as a private/custom fork, that’s a solid path for this kind of platform-specific need.

**Comment by lukemmtt** [comment]
Took @chungdaniel up on the fork suggestion and built a native C++ Windows plugin for this; PR is here: #290

It implements the full analytics engine in C++ (no Amplitude C/C++ SDK exists, so it's built from scratch): event queuing with persistent storage, WinHTTP transport, session tracking, device info from the Windows registry, app lifecycle events, etc. All 14 method channel calls match the iOS/Android/macOS behavior.

One caveat: the `platform` field shows `(none)` in Amplitude because the HTTP V2 API ignores it (it's server-computed). You can filter by `os_name = "Windows"` instead.

If you want to use it now, point your pubspec at the fork:

```yaml
amplitude_flutter:
  git:
    url: https://github.com/lukemmtt/Amplitude-Flutter.git
    ref: main
```

Details and platform comparison tables in the PR.

Thank you Amplitude team for doing what you do!
